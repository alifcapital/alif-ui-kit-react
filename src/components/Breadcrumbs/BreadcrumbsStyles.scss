@import '../../styles/index.scss';

.Breadcrumbs {
  .Breadcrumbs-list {
    display: flex;

    .Breadcrumbs-item {
      display: flex;

      .Breadcrumbs-label {
        cursor: pointer;
      }

      .Breadcrumbs-label-current {
        cursor: default;
        color: $black90;
      }

      .Breadcrumbs-separator {
        margin: 0 8px;
        display: flex;
        align-items: center;
      }
    }
  }
}
