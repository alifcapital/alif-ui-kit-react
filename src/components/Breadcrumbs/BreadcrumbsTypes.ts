import React from 'react';

export interface IBreadcrumbsItem {
  key: string | number;
  href: string;
  label: React.ReactNode | string;
}

export interface IBreadcrumbsProps {
  items: IBreadcrumbsItem[];
  onLabelClick?: (item: IBreadcrumbsItem) => void;
}
