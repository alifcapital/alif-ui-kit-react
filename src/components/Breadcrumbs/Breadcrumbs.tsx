import React from 'react';
import { RightNext } from 'alif-icon-kit-react';
import { IBreadcrumbsItem, IBreadcrumbsProps } from './BreadcrumbsTypes';
import './BreadcrumbsStyles.scss';
import clsx from 'clsx';

export const Breadcrumbs = (props: IBreadcrumbsProps) => {
  const { items, onLabelClick } = props;

  const handleLabelClick = (item: IBreadcrumbsItem) => {
    onLabelClick && onLabelClick(item);
    console.log('Вы нажали на: ', item);
  };

  return (
    <nav aria-label="Breadcrumb" className="Breadcrumbs">
      <ol className="Breadcrumbs-list">
        {items.map((item, index) => {
          const isCurrent = items.length - 1 === index;

          return (
            <li className="Breadcrumbs-item" key={index}>
              <div
                onClick={isCurrent ? undefined : () => handleLabelClick(item)}
                aria-current={isCurrent ? 'page' : undefined}
                className={clsx({
                  'Breadcrumbs-label': true,
                  'Breadcrumbs-label-current': isCurrent,
                })}
              >
                {item.label}
              </div>

              {!isCurrent && (
                <div aria-hidden="true" className="Breadcrumbs-separator">
                  <RightNext height="16" width="16" />
                </div>
              )}
            </li>
          );
        })}
      </ol>
    </nav>
  );
};
