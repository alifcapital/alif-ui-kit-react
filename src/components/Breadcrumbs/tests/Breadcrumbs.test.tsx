import React from 'react';
import { render } from '@testing-library/react';
import { Breadcrumbs } from '../Breadcrumbs';
import { IBreadcrumbsItem } from '../BreadcrumbsTypes';

const breadcrumbs = [
  {
    href: '/',
    label: <a href="https://alifshop.tj">Home</a>,
  },
  {
    href: '/react/components',
    label: 'Components',
  },
  {
    href: '/test',
    label: 'Breadcrumbs',
  },
];

describe('Breadcrumbs', () => {
  const handleClick = (item: IBreadcrumbsItem) => {
    window.location.href = item.href;
  };

  it('should match snapshot', () => {
    const { container } = render(<Breadcrumbs onLabelClick={handleClick} items={breadcrumbs} />);

    expect(container).toMatchSnapshot();
  });

  it('renders the proper accessibility attributes for nav tag', () => {
    const { container } = render(<Breadcrumbs onLabelClick={handleClick} items={breadcrumbs} />);
    const breadcrumbsElement = container.querySelector('.Breadcrumbs');

    expect(breadcrumbsElement).toHaveAttribute('aria-label', 'Breadcrumb');
  });

  it('should apply proper aria attributes to separators', () => {
    const { container } = render(<Breadcrumbs onLabelClick={handleClick} items={breadcrumbs} />);
    const separatorElement = container.querySelector('.Breadcrumbs-separator');

    expect(separatorElement).toHaveAttribute('aria-hidden', 'true');
  });

  it('should apply proper aria attributes to current link', () => {
    const { container } = render(
      <Breadcrumbs
        onLabelClick={handleClick}
        items={[
          {
            href: '/test',
            label: 'Breadcrumbs',
          },
        ]}
      />,
    );
    const labelElement = container.querySelector('.Breadcrumbs-label');

    expect(labelElement).toHaveAttribute('aria-current', 'page');
  });
});
