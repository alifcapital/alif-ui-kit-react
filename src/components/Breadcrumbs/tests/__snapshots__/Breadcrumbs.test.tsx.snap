// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs should match snapshot 1`] = `
<div>
  <nav
    aria-label="Breadcrumb"
    class="Breadcrumbs"
  >
    <ol
      class="Breadcrumbs-list"
    >
      <li
        class="Breadcrumbs-item"
      >
        <div
          class="Breadcrumbs-label"
        >
          <a
            href="https://alifshop.tj"
          >
            Home
          </a>
        </div>
        <div
          aria-hidden="true"
          class="Breadcrumbs-separator"
        >
          <svg
            fill="none"
            height="16"
            viewBox="0 0 25 25"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10.0615 6.91846L16.0615 12.9185L10.0615 18.9185"
              stroke="currentColor"
              stroke-linecap="square"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </div>
      </li>
      <li
        class="Breadcrumbs-item"
      >
        <div
          class="Breadcrumbs-label"
        >
          Components
        </div>
        <div
          aria-hidden="true"
          class="Breadcrumbs-separator"
        >
          <svg
            fill="none"
            height="16"
            viewBox="0 0 25 25"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10.0615 6.91846L16.0615 12.9185L10.0615 18.9185"
              stroke="currentColor"
              stroke-linecap="square"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </div>
      </li>
      <li
        class="Breadcrumbs-item"
      >
        <div
          aria-current="page"
          class="Breadcrumbs-label Breadcrumbs-label-current"
        >
          Breadcrumbs
        </div>
      </li>
    </ol>
  </nav>
</div>
`;
